Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/README.md

```diff
@@ -0,0 +1,907 @@
+Next, we have to setup metrics collection for DCGM and Triton Server.
+Metrics provide insight to the Kubernetes Horizontal Pod Autoscaler service and enable it to make autoscaling decisions based
+on the utilization and availability of deployed models.
+
+#### Create a Monitoring Namespace
+
+Create the `monitoring` namespace in your cluster for all of the metrics and monitoring services.
+
+1.  Run the command below to create the namespace.
+
+    ```bash
+    kubectl create namespace monitoring
+    ```
+
+#### Prometheus Services
+
+We a service to collect, store, aggregate, and provide metrics collected from your cluster and the services deployed in it.
+One of the easiest ways to do this is to leverage the functionality of the [Prometheus Metrics Server](https://prometheus.io/).
+Using the following steps, we'll install the Prometheus Stack for Kubernetes Helm chart so that we can leverage Prometheus.
+
+1.  Add the Prometheus Community chart repository to the local cache.
+
+    ```bash
+    helm repo add prometheus-community https://prometheus-community.github.io/helm-charts \
+      && helm repo update
+    ```
+
+2.  Run the command below to install the Prometheus Kubernetes Stack Helm chart.
+
+    ```bash
+    helm install -n monitoring prometheus prometheus-community/kube-prometheus-stack \
+      --set tolerations[0].key=nvidia.com/gpu \
+      --set tolerations[0].operator=Exists \
+      --set tolerations[0].effect=NoSchedule
+    ```
+
+#### NVIDIA Data Center GPU Manager (DCGM) Exporter
+
+The best solution for management of GPUs in your cluster is
+[NVIDIA DCGM](https://docs.nvidia.com/data-center-gpu-manager-dcgm)(DCGM).
+However, for this example we do not need the entirety of the DCGM stack.
+Instead, we'll use the steps below to install the [DCGM Exporter](https://github.com/NVIDIA/dcgm-exporter) to enable the
+collection of GPU metrics in your cluster.
+
+1.  Add the NVIDIA DCGM chart repository to the local cache.
+
+    ```bash
+    helm repo add nvidia-dcgm https://nvidia.github.io/dcgm-exporter/helm-charts \
+      && helm repo update
+    ```
+
+2.  Use the YAML contents below to create a file named `nvidia_dcgm-exporter_values.yaml`.
+
+    > [nvidia_dcgm-exporter_values.yaml](nvidia_dcgm-exporter_values.yaml)
+
+    The contents above were generated using `helm show values nvidia-dcgm/dcgm-exporter` and then modified specifically for
+    Triton Server autoscaling.
+
+4.  Install the DCGM Exporter Helm chart using the following command.
+
+    ```bash
+    helm install -n monitoring dcgm-exporter nvidia-dcgm/dcgm-exporter --values nvidia_dcgm-exporter_values.yaml
+    ```
+
+#### Connect DCGM and Triton Metrics to Prometheus
+
+We need to provide a mechanism that will scrape the metrics produced by DCGM Exporter and inject them into the Prometheus
+metrics server.
+The steps below will setup a Prometheus Adapter that collects metrics from every DCGM Exporter worker and provides them to
+Prometheus.
```

Comment:
You're right, not the Kubernetes metrics server -- but the adapter makes it so that you can query Prometheus through the Kubernetes Metrics API (which is what the HPA uses). It doesn't affect what DCGM provides to Prometheus, only what Prometheus provides to the HPA